app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt

```diff
@@ -217,8 +218,24 @@ class BrowserRobot {
     fun clickCopyImageLocation(): ViewInteraction = copyImageLocation.perform(click())
 
     fun clickLinkMatchingText(expectedText: String) {
-        mDevice.findObject(UiSelector().textContains(expectedText)).waitForExists(waitingTime)
-        mDevice.findObject(UiSelector().textContains(expectedText)).also { it.click() }
+        for (i in 1..RETRY_COUNT) {
+            try {
+                mDevice.findObject(UiSelector().textContains(expectedText)).waitForExists(waitingTime)
```

Comment:
you could also use waitingTimeLong if you think it helps

Reply:
Yes, thought about it, but I'll leave it with the standard waitingTime, don't think it would make a difference.